### PR TITLE
add search box and the ability to search

### PR DESCRIPTION
**Pull request by jorolf**
Prereqs:
- [x] https://github.com/ppy/osu-framework/pull/700

### CODE REVIEW

File: osu.Game/Overlays/Options/OptionsSection.cs [hunk 1]
```diff
@@ -1,4 +1,6 @@
-﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+﻿using System;
+using System.Collections.Generic;
```
Comment (smoogipooo) [inline osu.Game/Overlays/Options/OptionsSection.cs]: Move usings below license header.

**Comment by smoogipooo** [comment]
There's gotta be a prereq for this somewhere... Not compiling due to SearchContainer not being a FillFlowContainer (error on `Direction = FillDirection.Vertical`). Can you link it in the OP? Or if there's no prereq maybe you haven't committed the change.

**Comment by smoogipooo** [comment]
Added prereq to OP.

File: osu.Game/Overlays/Options/OptionsSection.cs (hunk 1 above)
Comment (Jorolf) [inline osu.Game/Overlays/Options/OptionsSection.cs]: how did that happen >.>

**Review by smoogipoo — APPROVED** [review]
This looks okay, I'm happy with this. Just needs CI fixes.

**Comment by smoogipooo** [comment]
I... Spoke too soon...

**Comment by peppy** [comment]
ty

**Pull request merged**